spec/spec_helper.rb

```diff
@@ -21,3 +21,36 @@ def read_attribute_for_validation(key)
     @attributes[key]
   end
 end
+
+class TestDynamicDomainModel
+  def self.where(*); end
+
+  def self.column_names
+    [domain_attribute].compact
+  end
+
+  def self.exists?(hash)
+    value = hash[self.domain_attribute.to_sym]
+    return false if value.nil?
+
+    existng_array = self.domain_attribute_values
+    existng_array.include?(value)
+  end
+
+  def self.domain_attribute
+    @domain_attribute ||= "domain"
+  end
+
+  def self.domain_attribute_values
+    @domain_attribute_values ||= []
+  end
+
+  def self.domain_attribute=(new_domain_attribute)
+    @domain_attribute = new_domain_attribute
+    @domain_attribute_values = domain_attribute_values
+  end
+
+  def self.domain_attribute_values=(new_domain_attribute_values)
+    @domain_attribute_values = new_domain_attribute_values
+  end
+end
```

Comment:
This is kind of a fake class we can use to test the hash for setting an active record query